js/client/modules/@arangodb/process-utils.js

```diff
@@ -1096,9 +1101,13 @@ function checkArangoAlive (arangod, options) {
       )
        ) {
       arangod.exitStatus = res;
-      analyzeServerCrash(arangod, options, 'health Check  - ' + res.signal);
+      msg = 'health Check Signal(' + res.signal + ') ';
+      analyzeServerCrash(arangod, options, msg);
       serverCrashedLocal = true;
-      print(Date() + " checkArangoAlive: Marking crashy - " + JSON.stringify(arangod));
+      arangod.message += msg;
+      msg = " checkArangoAlive: Marking crashy";
+      arangod.message += msg;
+      print(Date() + msg + ' - ' + JSON.stringify(arangod));
```

Comment:
Why `msg` is reused here?

Initially we will print `Date() + ' ArangoD of role [' + arangod.role + '] with PID ' + arangod.pid + ' is gone' + ':'`.

`arangod.message` finally will be
```
' ArangoD of role [' + arangod.role + '] with PID ' + arangod.pid + ' is gone ' + 'health Check Signal(' + res.signal + ') ' + " checkArangoAlive: Marking crashy"
```
or
```
'\n ' + 'health Check Signal(' + res.signal + ') ' + " checkArangoAlive: Marking crashy"
```

Afterwards we print `Date() + msg + ' - ' + JSON.stringify(arangod)`. `arangod` here will have `' ArangoD of role [' + arangod.role + '] with PID ' + arangod.pid + ' is gone ' + 'health Check Signal(' + res.signal + ') ' + " checkArangoAlive: Marking crashy"`.

What is the logic of interpretation?

Reply:
simply msg is re-used. But its re-assigned a value, so its content doesn't add up.